src/globalInfo/globalInfoConfig.ts

```diff
@@ -0,0 +1,114 @@
+ */
+import { AnyJson, isPlainObject, JsonMap, Optional } from '@salesforce/ts-types';
+import { Global } from '../global';
+import { ConfigFile } from '../config/configFile';
+import { ConfigValue } from '../config/configStore';
+import { SfdxDataHandler } from './sfdxDataHandler';
+import { OrgAccessor } from './accessors/orgAccessor';
+import { TokenAccessor } from './accessors/tokenAccessor';
+import { AliasAccessor } from './accessors/aliasAccessor';
+import { SfInfo, SfInfoKeys } from './types';
+
+export function deepCopy<T extends AnyJson>(data: T): T {
+  return JSON.parse(JSON.stringify(data)) as T;
+}
+
+export class GlobalInfo extends ConfigFile<ConfigFile.Options, SfInfo> {
+  protected static encryptedKeys = [/token/gi, /password/gi, /secret/gi];
+  private static EMPTY_DATA_MODEL: SfInfo = {
+    [SfInfoKeys.ORGS]: {},
+    [SfInfoKeys.TOKENS]: {},
+    [SfInfoKeys.ALIASES]: {},
+  };
+  private static instance: Optional<GlobalInfo>;
+
+  private sfdxHandler = new SfdxDataHandler();
+
+  public static get emptyDataModel(): SfInfo {
+    return deepCopy<SfInfo>(GlobalInfo.EMPTY_DATA_MODEL);
+  }
+
+  public static async getInstance(): Promise<GlobalInfo> {
+    if (!GlobalInfo.instance) {
+      GlobalInfo.instance = await GlobalInfo.create();
+    }
+    return GlobalInfo.instance;
+  }
+
+  /**
+   * Clear the cache to force reading from disk.
+   *
+   * *NOTE: Only call this method if you must and you know what you are doing.*
+   */
+  public static clearInstance(): void {
+    delete GlobalInfo.instance;
+  }
+
+  public static getFileName(): string {
+    return 'sf.json';
+  }
+
+  /**
+   * Gets default options for the SfConfig
+   */
+  public static getDefaultOptions(): ConfigFile.Options {
+    return {
+      isGlobal: true,
+      isState: true,
+      filename: GlobalInfo.getFileName(),
+      stateFolder: Global.SF_STATE_FOLDER,
+    };
+  }
+
+  public get orgs(): OrgAccessor {
+    return new OrgAccessor(this);
+  }
+
+  public get tokens(): TokenAccessor {
+    return new TokenAccessor(this);
+  }
+
+  public get aliases(): AliasAccessor {
+    return new AliasAccessor(this);
+  }
+
+  public set(key: string, value: ConfigValue): void {
+    if (isPlainObject(value)) {
+      value = this.timestamp(value as JsonMap);
+    }
+    super.set(key, value);
+  }
+
+  public async write(newContents?: SfInfo): Promise<SfInfo> {
+    const result = await super.write(newContents);
+    if (Global.SFDX_INTEROPERABILITY) await this.sfdxHandler.write(result);
+    return result;
+  }
+
+  protected async init(): Promise<void> {
+    await this.initCrypto();
+    const contents = Global.SFDX_INTEROPERABILITY ? await this.mergeWithSfdxData() : await this.loadSfData();
+    this.setContents(contents);
+    await this.write(contents);
+  }
+
+  private timestamp<T extends JsonMap>(data: T): T {
+    return Object.assign(data, { timestamp: new Date() });
```

Comment:
A little simpler, [a little faster](https://jsben.ch/uzw0r), and [doesn't trigger setters](https://thecodebarbarian.com/object-assign-vs-object-spread.html#differences-versus-objectassign).
```suggestion
    return {...data, timestamp: new Date()};
```